src/include/firebird/iberror.h

```diff
@@ -17,32 +17,38 @@
 
 #ifdef __cplusplus /* c++ definitions */
 
-const ISC_STATUS isc_facility = 20;
-const ISC_STATUS isc_base = isc_facility << 24;
-const ISC_STATUS isc_factor = 1;
-
-const ISC_STATUS isc_arg_end			= 0;	// end of argument list
-const ISC_STATUS isc_arg_gds			= 1;	// generic DSRI status value
-const ISC_STATUS isc_arg_string			= 2;	// string argument
-const ISC_STATUS isc_arg_cstring		= 3;	// count & string argument
-const ISC_STATUS isc_arg_number			= 4;	// numeric argument (long)
-const ISC_STATUS isc_arg_interpreted	= 5;	// interpreted status code (string)
-const ISC_STATUS isc_arg_vms			= 6;	// VAX/VMS status code (long)
-const ISC_STATUS isc_arg_unix			= 7;	// UNIX error code
-const ISC_STATUS isc_arg_domain			= 8;	// Apollo/Domain error code
-const ISC_STATUS isc_arg_dos			= 9;	// MSDOS/OS2 error code
-const ISC_STATUS isc_arg_mpexl			= 10;	// HP MPE/XL error code
-const ISC_STATUS isc_arg_mpexl_ipc		= 11;	// HP MPE/XL IPC error code
-const ISC_STATUS isc_arg_next_mach		= 15;	// NeXT/Mach error code
-const ISC_STATUS isc_arg_netware		= 16;	// NetWare error code
-const ISC_STATUS isc_arg_win32			= 17;	// Win32 error code
-const ISC_STATUS isc_arg_warning		= 18;	// warning argument
-const ISC_STATUS isc_arg_sql_state		= 19;	// SQLSTATE
+#if __cplusplus >= 201703L // C++17 or later
+#define STATUS_CONST inline constexpr ISC_STATUS
```

Comment:
This is public header, so I think some FB-specific prefix in the macro is good.
Also, it should be good to undefine it after usage.